src/components/app/data/queries/queries.js

```diff
@@ -268,6 +268,25 @@ export function queryCanRedeem(enterpriseUuid, courseMetadata, isEnrollableBuffe
     ._ctx.canRedeem(availableCourseRunKeys);
 }
 
+/**
+ * Helper function to assist querying with React Query package
+ *
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.course
+ * ._ctx.canRedeem(availableCourseRunKeys)
+ * @returns {Types.QueryOptions}
+ */
+export function queryCanUpgradeWithLearnerCredit(enterpriseUuid, courseRunKey) {
+  console.log(enterpriseUuid, courseRunKey);
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.course(courseRunKey)
```

Comment:
[suggestion] Related to the other comment about whether we could re-use the existing `canRedeem` query key from `queries`, I suppose we'd have to pass `null` here at least until the `enterprise_course_enrollments` API begins returning the top-level course key in its response.

The `._ctx.course(...)` query key expects a top-level course key, so it might not be semantically correct to pass a course run key here, is my concern.

Especially knowing we'd eventually have a top-level course key to use here (based on a future ticket), I think it'd make sense to keep the query key standardized to the course vs. course run structure we have today with the existing `canRedeem` query from `queries` 🤔